### PR TITLE
type_def argument of Client is now deprecated.

### PR DESCRIPTION
**Pull request by leszekhanusz**
The Client of gql allows us now to pass a local schema either:

- with the `schema` argument if the schema is a GraphQLSchema type
- with the `type_def` argument if the schema is passed as a string

But this is basically the same thing expressed with different types.

I propose to simplify the documentation that the `type_def` argument becomes deprecated and that the local schema is passed using the `schema` argument for both types.

To summarize, you should use `schema=` now instead of previously `type_def=` in the Client.

The schema argument will accept either:
- a string (previously passed to the type_def argument)
- a GraphQLSchema (like before)

### CODE REVIEW

File: gql/client.py [hunk 1]
```diff
@@ -31,22 +32,33 @@ def __init__(
         assert not (
             type_def and introspection
         ), "Cannot provide introspection type definition at the same time."
```
Comment (KingDarBoja) [inline gql/client.py:34]: Not related to this PR but I think there is a missing connector `and` on the message.

Should be:
`Cannot provide introspection and type definition at the same time.`

File: gql/client.py [hunk 2]
```diff
@@ -31,22 +32,33 @@ def __init__(
...
         if introspection:
             assert (
                 not schema
             ), "Cannot provide introspection and schema at the same time."
             schema = build_client_schema(introspection)
-        elif type_def:
+
+        if isinstance(schema, str):
+            type_def_ast = parse(schema)
+            schema = cast(GraphQLSchema, build_ast_schema(type_def_ast))
```
Comment (KingDarBoja) [inline gql/client.py:54]: Why the cast? As far as I know, the [**build_ast_schema**](https://github.com/graphql-python/graphql-core/blob/3267c4c5c5ea06968beb59f16425de4a4b835ee1/src/graphql/utilities/build_ast_schema.py#L23) returns a `GraphQLSchema` type.

Comment (leszekhanusz) [reply gql/client.py]: I had a mypy error earlier. But now it seems it is not necessary anymore. Well spotted.

File: gql/client.py (hunk 1 above)
Comment (leszekhanusz) [reply gql/client.py]: Indeed, I made the correction

**Comment by leszekhanusz** [comment]
> I would rather have removed the `type_def` argument and mark as breaking change for gql v3.

We could leave it as deprecated now and remove it once we are at the stable 3.0 ?

**Comment by KingDarBoja** [comment]
> > I would rather have removed the `type_def` argument and mark as breaking change for gql v3.
> 
> We could leave it as deprecated now and remove it once we are at the stable 3.0 ?

Well I am okay with that as long as it doesn't stay for more than 2 minor releases after stable 😄
